### ISSUE TITLE
Bucket name too long: bkt-prj-bu1-p-sample-base-bx0b-southamerica-east1-cmek-encrypted-axlhk

### ISSUE THREAD

**Issue by qingvincentyin** [terraform-google-modules/terraform-example-foundation#982]
### TL;DR

I'm using the `southamerica-east1` region. So, I set the content of [production.auto.example.tfvars](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/production.auto.example.tfvars) to this:

```terraform
# File: 4-projects/production.auto.example.tfvars
...
location_gcs = "SOUTHAMERICA-EAST1"
```

The resulting bucket name would be `bkt-prj-bu1-p-sample-base-bx0b-southamerica-east1-cmek-encrypted-axlhk` which is longer than the 63-character limit for a bucket name.

### Expected behavior

The CICD pipeline in `prj-b-cicd` should succeed upon `git push`.

### Observed behavior

The Cloud Build in `prj-b-cicd` gives this:

Already have image (with digest): southamerica-east1-docker.pkg.dev/prj-b-cicd-3w96/tf-runners/terraform:v1
*************** TERRAFORM APPLY *******************
      At environment: business_unit_1/shared 
***************************************************
pwd=/workspace/business_unit_1/shared
+ terraform apply -input=false -auto-approve /workspace/tmp_plan/business_unit_1-shared.tfplan

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

apply_triggers_id = [
  "projects/prj-bu1-c-infra-pipeline-txgk/triggers/99448e90-e84d-480d-9559-056e9f1f032e",
]
artifact_buckets = {
  "bu1-example-app" = "bkt-prj-bu1-c-infra-pipeline-txgk-bu1-example-app-artifacts"
}
cloudbuild_project_id = "prj-bu1-c-infra-pipeline-txgk"
default_region = "southamerica-east1"
enable_cloudbuild_deploy = true
log_buckets = {
  "bu1-example-app" = "bkt-prj-bu1-c-infra-pipeline-txgk-bu1-example-app-logs"
}
plan_triggers_id = [
  "projects/prj-bu1-c-infra-pipeline-txgk/triggers/58859832-c495-467c-bd32-2595f3ad5e20",
]
repos = toset([
  "bu1-example-app",
])
state_buckets = {
  "bu1-example-app" = "bkt-prj-bu1-c-infra-pipeline-txgk-bu1-example-app-state"
}
terraform_service_accounts = {
  "bu1-example-app" = "sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com"
}
+ set +x
*************** TERRAFORM APPLY *******************
      At environment: business_unit_1/production 
***************************************************
pwd=/workspace/business_unit_1/production
+ terraform apply -input=false -auto-approve /workspace/tmp_plan/business_unit_1-production.tfplan
module.env.random_string.bucket_name: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.random_id.random_project_id_suffix: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.random_string.random_project_id_suffix[0]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.random_id.random_project_id_suffix: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.random_id.random_project_id_suffix: Creation complete after 0s [id=OW8]
module.env.module.env_secrets_project.module.project.module.project-factory.random_id.random_project_id_suffix: Creation complete after 0s [id=4KE]
module.env.random_string.bucket_name: Creation complete after 0s [id=axlhk]
module.env.module.env_secrets_project.module.project.module.project-factory.random_string.random_project_id_suffix[0]: Creation complete after 0s [id=rasp]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.random_string.random_project_id_suffix[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.random_string.random_project_id_suffix[0]: Creation complete after 0s [id=bx0b]
module.env.module.base_shared_vpc_project.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets["bu1-example-app-sb-p-shared-base-northamerica-northeast1-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com"]: Creating...
module.env.module.base_shared_vpc_project.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets["bu1-example-app-sb-p-shared-base-southamerica-west1-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com"]: Creating...
module.env.module.base_shared_vpc_project.google_folder_iam_member.folder_network_viewer["bu1-example-app"]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Creating...
module.env.module.base_shared_vpc_project.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets["bu1-example-app-sb-p-shared-base-southamerica-west1-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com"]: Creation complete after 5s [id=projects/prj-p-shared-base-wf14/regions/southamerica-west1/subnetworks/sb-p-shared-base-southamerica-west1/roles/compute.networkUser/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.google_folder_iam_member.folder_network_viewer["bu1-example-app"]: Creation complete after 6s [id=folders/889115438821/roles/compute.networkViewer/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets["bu1-example-app-sb-p-shared-base-northamerica-northeast1-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com"]: Creation complete after 9s [id=projects/prj-p-shared-base-wf14/regions/northamerica-northeast1/subnetworks/sb-p-shared-base-northamerica-northeast1/roles/compute.networkUser/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [10s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [20s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [20s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [30s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [30s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [40s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [40s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [50s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [50s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Still creating... [1m0s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Still creating... [1m0s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project.main: Creation complete after 1m3s [id=projects/prj-bu1-p-sample-base-bx0b]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["iam.googleapis.com"]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_service_account.default_service_account[0]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.google_project.main: Creation complete after 1m4s [id=projects/prj-bu1-p-env-secrets-rasp]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudkms.googleapis.com"]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["secretmanager.googleapis.com"]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.google_service_account.default_service_account[0]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["logging.googleapis.com"]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_service_account.default_service_account[0]: Creation complete after 2s [id=projects/prj-bu1-p-sample-base-bx0b/serviceAccounts/project-service-account@prj-bu1-p-sample-base-bx0b.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets[1]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets[0]: Creating...
module.env.module.env_secrets_project.module.project.module.project-factory.google_service_account.default_service_account[0]: Creation complete after 2s [id=projects/prj-bu1-p-env-secrets-rasp/serviceAccounts/project-service-account@prj-bu1-p-env-secrets-rasp.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets[1]: Creation complete after 6s [id=projects/prj-p-shared-base-wf14/regions/southamerica-west1/subnetworks/sb-p-shared-base-southamerica-west1/roles/compute.networkUser/serviceAccount:project-service-account@prj-bu1-p-sample-base-bx0b.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["iam.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.service_account_role_to_vpc_subnets[0]: Creation complete after 9s [id=projects/prj-p-shared-base-wf14/regions/northamerica-northeast1/subnetworks/sb-p-shared-base-northamerica-northeast1/roles/compute.networkUser/serviceAccount:project-service-account@prj-bu1-p-sample-base-bx0b.iam.gserviceaccount.com]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudkms.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["secretmanager.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["logging.googleapis.com"]: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["iam.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["secretmanager.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudkms.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["logging.googleapis.com"]: Still creating... [20s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["iam.googleapis.com"]: Creation complete after 22s [id=prj-bu1-p-sample-base-bx0b/iam.googleapis.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Creation complete after 22s [id=prj-bu1-p-sample-base-bx0b/cloudresourcemanager.googleapis.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Creation complete after 22s [id=prj-bu1-p-sample-base-bx0b/billingbudgets.googleapis.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.apis_service_account_role_to_vpc_subnets[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project_default_service_accounts.default_service_accounts[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_shared_vpc_service_project.shared_vpc_attachment[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.apis_service_account_role_to_vpc_subnets[1]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_project_default_service_accounts.default_service_accounts[0]: Creation complete after 0s [id=projects/prj-bu1-p-sample-base-bx0b]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["billingbudgets.googleapis.com"]: Creation complete after 23s [id=prj-bu1-p-env-secrets-rasp/billingbudgets.googleapis.com]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["secretmanager.googleapis.com"]: Creation complete after 23s [id=prj-bu1-p-env-secrets-rasp/secretmanager.googleapis.com]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["cloudkms.googleapis.com"]: Creation complete after 23s [id=prj-bu1-p-env-secrets-rasp/cloudkms.googleapis.com]
module.env.module.env_secrets_project.module.project.module.project-factory.module.project_services.google_project_service.project_services["logging.googleapis.com"]: Creation complete after 23s [id=prj-bu1-p-env-secrets-rasp/logging.googleapis.com]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project_default_service_accounts.default_service_accounts[0]: Creating...
module.env.module.env_secrets_project.module.project.module.budget.data.google_project.project[0]: Reading...
module.env.module.kms.google_kms_key_ring.key_ring: Creating...
module.env.module.kms.google_kms_key_ring.key_ring: Creation complete after 0s [id=projects/prj-bu1-p-env-secrets-rasp/locations/southamerica-east1/keyRings/sample-keyring]
module.env.module.kms.google_kms_crypto_key.key_ephemeral[0]: Creating...
module.env.module.kms.google_kms_crypto_key.key_ephemeral[0]: Creation complete after 0s [id=projects/prj-bu1-p-env-secrets-rasp/locations/southamerica-east1/keyRings/sample-keyring/cryptoKeys/crypto-key-example]
module.env.module.env_secrets_project.module.project.module.project-factory.google_project_default_service_accounts.default_service_accounts[0]: Creation complete after 0s [id=projects/prj-bu1-p-env-secrets-rasp]
module.env.module.env_secrets_project.module.project.module.budget.data.google_project.project[0]: Read complete after 1s [id=projects/prj-bu1-p-env-secrets-rasp]
module.env.module.env_secrets_project.module.project.module.budget.google_billing_budget.budget[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.apis_service_account_role_to_vpc_subnets[1]: Creation complete after 5s [id=projects/prj-p-shared-base-wf14/regions/southamerica-west1/subnetworks/sb-p-shared-base-southamerica-west1/roles/compute.networkUser/serviceAccount:1077561455172@cloudservices.gserviceaccount.com]
module.env.module.env_secrets_project.module.project.module.budget.google_billing_budget.budget[0]: Creation complete after 2s [id=billingAccounts/014410-8FC3C2-F86263/budgets/0bde6dd0-1c70-42be-8716-c6ec3fc939b8]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_subnetwork_iam_member.apis_service_account_role_to_vpc_subnets[0]: Creation complete after 8s [id=projects/prj-p-shared-base-wf14/regions/northamerica-northeast1/subnetworks/sb-p-shared-base-northamerica-northeast1/roles/compute.networkUser/serviceAccount:1077561455172@cloudservices.gserviceaccount.com]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_shared_vpc_service_project.shared_vpc_attachment[0]: Still creating... [10s elapsed]
module.env.module.base_shared_vpc_project.module.project.module.project-factory.google_compute_shared_vpc_service_project.shared_vpc_attachment[0]: Creation complete after 12s [id=prj-p-shared-base-wf14/prj-bu1-p-sample-base-bx0b]
module.env.module.base_shared_vpc_project.module.project.module.budget.data.google_project.project[0]: Reading...
module.env.data.google_storage_project_service_account.gcs_account: Reading...
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/compute.instanceAdmin.v1"]: Creating...
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/iam.serviceAccountUser"]: Creating...
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/iam.serviceAccountAdmin"]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.budget.data.google_project.project[0]: Read complete after 1s [id=projects/prj-bu1-p-sample-base-bx0b]
module.env.module.base_shared_vpc_project.module.project.module.budget.google_billing_budget.budget[0]: Creating...
module.env.data.google_storage_project_service_account.gcs_account: Read complete after 1s [id=service-1077561455172@gs-project-accounts.iam.gserviceaccount.com]
module.env.module.kms.google_kms_crypto_key_iam_binding.decrypters[0]: Creating...
module.env.module.kms.google_kms_crypto_key_iam_binding.encrypters[0]: Creating...
module.env.module.base_shared_vpc_project.module.project.module.budget.google_billing_budget.budget[0]: Creation complete after 2s [id=billingAccounts/014410-8FC3C2-F86263/budgets/860271a0-e465-4374-9ce0-e28fd70d0159]
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/iam.serviceAccountAdmin"]: Creation complete after 8s [id=prj-bu1-p-sample-base-bx0b/roles/iam.serviceAccountAdmin/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/compute.instanceAdmin.v1"]: Creation complete after 9s [id=prj-bu1-p-sample-base-bx0b/roles/compute.instanceAdmin.v1/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.base_shared_vpc_project.google_project_iam_member.app_infra_pipeline_sa_roles["bu1-example-app-sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com-roles/iam.serviceAccountUser"]: Creation complete after 9s [id=prj-bu1-p-sample-base-bx0b/roles/iam.serviceAccountUser/serviceAccount:sa-tf-cb-bu1-example-app@prj-bu1-c-infra-pipeline-txgk.iam.gserviceaccount.com]
module.env.module.kms.google_kms_crypto_key_iam_binding.encrypters[0]: Creation complete after 9s [id=projects/prj-bu1-p-env-secrets-rasp/locations/southamerica-east1/keyRings/sample-keyring/cryptoKeys/crypto-key-example/roles/cloudkms.cryptoKeyEncrypter]
module.env.module.kms.google_kms_crypto_key_iam_binding.decrypters[0]: Creation complete after 9s [id=projects/prj-bu1-p-env-secrets-rasp/locations/southamerica-east1/keyRings/sample-keyring/cryptoKeys/crypto-key-example/roles/cloudkms.cryptoKeyDecrypter]
module.env.module.gcs_buckets.google_storage_bucket.bucket: Creating...
╷
│ Error: error: bucket name validation failed bkt-prj-bu1-p-sample-base-bx0b-southamerica-east1-cmek-encrypted-axlhk
│ 
│   with module.env.module.gcs_buckets.google_storage_bucket.bucket,
│   on .terraform/modules/env.gcs_buckets/modules/simple_bucket/main.tf line 17, in resource "google_storage_bucket" "bucket":
│   17: resource "google_storage_bucket" "bucket" {
│ 
╵
+ exit 1

### Terraform Configuration

```hcl
It's your own terraform.
```

### Terraform Version

```sh
It's the terraform version packaged to the project `prj-b-cicd`.
```
